snapshot_manager/tests/github_util_test.py

```diff
@@ -1,125 +1,749 @@
-        self.assertEqual(issue.get_labels().totalCount, 0)
-
-        # Ensure those labels that we need for our test exist
-        chroot = "fedora-rawhide-x86_64"
-        all_chroots = [chroot]
-        logging.info("Creating test labels")
-        gh.create_labels_for_in_testing(all_chroots)
-        gh.create_labels_for_tests_failed_on(all_chroots)
-        gh.create_labels_for_tested_on(all_chroots)
-
-        in_testing = gh.label_in_testing(chroot=chroot)
-        failed_on = gh.label_failed_on(chroot=chroot)
-        tested_on = gh.label_tested_on(chroot=chroot)
-
-        all_test_states = [in_testing, failed_on, tested_on]
-        for test_state in all_test_states:
-            logging.info(f"Flipping test label for chroot {chroot} to: {test_state}")
-            gh.flip_test_label(issue, chroot, test_state)
-            labels = issue.get_labels()
-            self.assertIsNotNone(labels)
-            self.assertEqual(labels.totalCount, 1)
-            page = labels.get_page(0)
-            self.assertIsNotNone(page)
-            self.assertEqual(page[0].name, test_state)
-        pass
-
-    def test_get_workflow(self):
-        try:
-            gh = github_util.GithubClient(config=self.config)
-        except github_util.MissingToken:
-            pytest.skip(
-                "Skip test because this execution doesn't have access to a Github token"
-            )
+    assert str(ex.value) == f"invalid comment object passed: {obj}"
+
+
+@pytest.mark.parametrize(
+    "input, expected, func",
+    [
+        (
+            "fedora-rawhide-s390x",
+            "in_testing/fedora-rawhide-s390x",
+            lambda gh: gh.label_in_testing,
+        ),
+        (
+            "fedora-rawhide-x86_64",
+            "tests_failed_on/fedora-rawhide-x86_64",
+            lambda gh: gh.label_failed_on,
+        ),
+        (
+            "fedora-rawhide-ppc64le",
+            "tested_on/fedora-rawhide-ppc64le",
+            lambda gh: gh.label_tested_on,
+        ),
+    ],
+)
+def test_label(input, expected, func, github_client_fxt):
+    assert func(github_client_fxt)(chroot=input) == expected
+
+
+def test_flip_test_label(github_client_fxt):
+    issue_mock = mock.MagicMock(spec=github.Issue.Issue)
+    issue_mock.add_to_labels = mock.MagicMock()
+
+    # Flip the test status of this chroot: fedora-rawhide-x86_64
+    chroot = "fedora-rawhide-x86_64"
+    new_label = f"tests_failed_on/{chroot}"
+
+    issue_mock.get_labels.return_value = [
+        MyLabel(name="project/clang"),
+        # This will be removed
+        MyLabel(name="in_testing/fedora-rawhide-x86_64"),
+        MyLabel(name="tested_on/fedora-rawhide-ppc64le"),
+    ]
+
+    github_client_fxt.flip_test_label(
+        issue=issue_mock, chroot="fedora-rawhide-x86_64", new_label=new_label
+    )
+
+    issue_mock.add_to_labels.assert_called_once_with(new_label)
+    issue_mock.remove_from_labels.assert_called_once_with(
+        "in_testing/fedora-rawhide-x86_64"
+    )
```

Comment:
Isn't it necessary to check if a single label got removed? Or is this last call doing exactly that?